Add benchmark for channel_shuffle operator

taivu1998 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43602 Implement ChannelShuffle op with XNNPACK
* **#43509 Add benchmark for channel_shuffle operator**

Differential Revision: [D23299972](https://our.internmc.facebook.com/intern/diff/D23299972)

Commit:
Add benchmark for channel_shuffle operator

[ghstack-poisoned]

kimishpatel reviewed (commented):
Can you run the test and report the results here?

taivu1998 commented:
| Hyperparameters  | Time (us) |
| ------------- | ------------- |
| Batch_size: 1, Channels per group: 16, Height: 16, Width: 16, Groups: 1  | 3.990  |
| Batch_size: 2, Channels per group: 16, Height: 16, Width: 16, Groups: 2  | 10.510 |
| Batch_size: 2, Channels per group: 32, Height: 32, Width: 32, Groups: 2  | 47.635 |
| Batch_size: 4, Channels per group: 32, Height: 32, Width: 32, Groups: 4  | 735.698 |
| Batch_size: 4, Channels per group: 64, Height: 64, Width: 64, Groups: 4  | 9783.544 |

kimishpatel reviewed (approved):
Looks good to me.

Commit:
Update on "Add benchmark for channel_shuffle operator"


Differential Revision: [D23299972](https://our.internmc.facebook.com/intern/diff/D23299972)

[ghstack-poisoned]

Commit:
Update on "Add benchmark for channel_shuffle operator"


Differential Revision: [D23299972](https://our.internmc.facebook.com/intern/diff/D23299972)

[ghstack-poisoned]

Commit:
Update on "Add benchmark for channel_shuffle operator"


Differential Revision: [D23299972](https://our.internmc.facebook.com/intern/diff/D23299972)

[ghstack-poisoned]

mingzhe09088 commented on benchmarks/operator_benchmark/pt/channel_shuffle_test.py:
```diff
@@ -0,0 +1,64 @@
+
+
+# Configs for PT channel_shuffle operator
+channel_shuffle_long_configs = op_bench.cross_product_configs(
```
Could you try to reduce the number of configs here? This config will create 5 x 4 x 4 x 4 x 4 x 2 = 2560 different test cases. Ideally we would like each operator to have less than 100 different tests.

kimishpatel replied:
Thanks @mingzhe09088. Tai, can you reduce each param to only two values?

mingzhe09088 commented:
Could you add this benchmark to https://github.com/pytorch/pytorch/blob/master/benchmarks/operator_benchmark/benchmark_all_other_test.py so it will be tested automatically?

Commit:
Update on "Add benchmark for channel_shuffle operator"


Differential Revision: [D23299972](https://our.internmc.facebook.com/intern/diff/D23299972)

[ghstack-poisoned]

Commit:
Update on "Add benchmark for channel_shuffle operator"


Differential Revision: [D23299972](https://our.internmc.facebook.com/intern/diff/D23299972)

[ghstack-poisoned]

Commit:
Update on "Add benchmark for channel_shuffle operator"


Differential Revision: [D23299972](https://our.internmc.facebook.com/intern/diff/D23299972)

[ghstack-poisoned]

Commit:
Update on "Add benchmark for channel_shuffle operator"


Differential Revision: [D23299972](https://our.internmc.facebook.com/intern/diff/D23299972)

[ghstack-poisoned]

Commit:
Update on "Add benchmark for channel_shuffle operator"


Differential Revision: [D23299972](https://our.internmc.facebook.com/intern/diff/D23299972)

[ghstack-poisoned]

Commit:
Update on "Add benchmark for channel_shuffle operator"


Differential Revision: [D23299972](https://our.internmc.facebook.com/intern/diff/D23299972)

[ghstack-poisoned]

Commit:
Update on "Add benchmark for channel_shuffle operator"


Differential Revision: [D23299972](https://our.internmc.facebook.com/intern/diff/D23299972)

[ghstack-poisoned]
